sympy/core/equation.py

```diff
@@ -0,0 +1,359 @@
+        elif isinstance(a, Equation) and isinstance(b, Equation):
+            return Equation(opfunc_ab(a.lhs, b.lhs), opfunc_ab(a.rhs, b.rhs))
+        else:
+            raise TypeError('One of a or b should be an equation')
+
+    def __add__(self, other):
+        return self._binary_op(self, other, lambda a, b: a + b)
+
+    def __radd__(self, other):
+        return self._binary_op(other, self, lambda a, b: a + b)
+
+    def __mul__(self, other):
+        return self._binary_op(self, other, lambda a, b: a * b)
+
+    def __rmul__(self, other):
+        return self._binary_op(other, self, lambda a, b: a * b)
+
+    def __sub__(self, other):
+        return self._binary_op(self, other, lambda a, b: a - b)
+
+    def __rsub__(self, other):
+        return self._binary_op(other, self, lambda a, b: a - b)
+
+    def __truediv__(self, other):
+        return self._binary_op(self, other, lambda a, b: a / b)
+
+    def __rtruediv__(self, other):
+        return self._binary_op(other, self, lambda a, b: a / b)
+
+    def __mod__(self, other):
+        return self._binary_op(self, other, lambda a, b: a % b)
+
+    def __rmod__(self,other):
+        return self._binary_op(other, self, lambda a, b: a % b)
+
+    def __pow__(self, other):
+        return self._binary_op(self, other, lambda a, b: a ** b)
+
+    def __rpow__(self, other):
+        return self._binary_op(other, self, lambda a, b: a ** b)
+
+    def _eval_power(self, other):
+        return self.__pow__(other)
+    
+#####
+# Output helper functions
+#####
+    def __repr__(self):
+        return(str(self.lhs)+self.relop+str(self.rhs))
+
+    def _latex(self,obj,**kwargs):
+        from sympy.printing import latex
+        return(latex(self.lhs)+self.relop+latex(self.rhs))
+
+    def __str__(self):
+        return(self.__repr__())
+
+#####
+# Operation helper functions
+#####
+    def expand(self, *args, **kwargs):
+        return Equation(self.lhs.expand(*args, **kwargs),self.rhs.expand(*args, **kwargs))
+
+    def simplify(self, *args, **kwargs):
+        return self._eval_simplify(*args, **kwargs)
+
+    def _eval_simplify(self, *args, **kwargs):
+        return Equation(self.lhs.simplify(*args, **kwargs),self.rhs.simplify(*args, **kwargs))
+
+    def _eval_factor(self, *args, **kwargs):
+        # TODO: cancel out factors common to both sides.
+        return Equation(self.lhs.factor(*args, **kwargs),self.rhs.factor(*args, **kwargs))
+
+    def factor(self, *args, **kwargs):
+        return self._eval_factor(*args, **kwargs)
+
+    def _eval_collect(self, *args, **kwargs):
+        from sympy.simplify.radsimp import collect
+        return Equation(collect(self.lhs, *args, **kwargs),collect(self.rhs, *args, **kwargs))
+
+    def collect(self, *args, **kwargs):
+        return self._eval_collect(*args, **kwargs)
+
+    def evalf(self, *args, **kwargs):
+        return Equation(self.lhs.evalf(*args, **kwargs),self.rhs.evalf(*args, **kwargs))
```

Comment:
There is `EvalfMixin` which should provide these methods (as it does to Expr).

Reply:
Using `EvalfMixin` does not seem to work correctly. The expected behavior which I get using the definition above is:
```
>>> eq2
p = R*T*n/V
>>> var('L atm mol K', positive=True, real=True) # units
(L, atm, mol, K)
>>> eq2.subs({R:0.08206*L*atm/mol/K,T:273*K,n:1.00*mol,V:24.0*L})
p = 0.9334325*atm
>>> eq2.subs({R:0.08206*L*atm/mol/K,T:273*K,n:1.00*mol,V:24.0*L}).evalf(4)
p = 0.9334*atm
```
Using `EvalfMixin`, I get:
```
>>> eq2.subs({R:0.08206*L*atm/mol/K,T:273*K,n:1.00*mol,V:24.0*L}).evalf(4)
p = 0.9334325*atm
>>> eq2.subs({R:0.08206*L*atm/mol/K,T:273*K,n:1.00*mol,V:24.0*L}).evalf(2)
p = 0.9334325*atm
```
Notice no rounding seems to be done. I believe the issue is that `EvalfMixin` is not `side` aware so skips the appropriate rounding of the `rhs`. I have also found that `evalf` does not seem to work with the `subs=` keyword on all the expressions I have tried. This suggests other issues with `evalf`.

As one of the uses I have for the equation class is to substitute values with units into the final equation in an algebraic chain the rounding is a key feature. That works if we use the specialized `Equation.evaf()`. For the time being I will leave it in.

Reply:
Maybe `EvalfMixin` does not make sense if the intention is not for the object to completely evaluate to a Float.

Reply:
I left `EvalfMixin` in along with my overrides so that if somebody does just numerical substitution the normal evaluation to float works as expected. I have a test to make sure that is the case.